### PR TITLE
Enabling Cobertura CC tool for Ant and Maven

### PR DESCRIPTION
**Pull request by prawalagarwal**
Description: Currently we support only JaCoCo as Code Coverage tools for Ant and Maven tasks. This code change adds the support of using Cobertura for these tasks.

Testing: Unit Testing and Manual Testing. Functional tests pending.

### CODE REVIEW

File: Tasks/ANT/ant.ps1 [hunk 1]
```diff
@@ -131,29 +136,64 @@ else
...
-		Write-Verbose "Calling Publish-CodeCoverage" -Verbose
-		Publish-CodeCoverage -CodeCoverageTool $codeCoverageTool -SummaryFileLocation $summaryFile -ReportDirectory $reportDirectory -Context $distributedTaskContext    
-   }
-   else
-   {
-        Write-Host "##vso[task.logissue type=warning;code=006003;]"
-		Write-Warning "No code coverage results found to be published. This could occur if there were no tests executed or there was a build failure. Check the ant output for details." -Verbose
+       # run report code coverage task which generates code coverage reports.
+	   $reportsGenerationFailed = $false
+	   Write-Verbose "Collecting code coverage reports" -Verbose
```
Comment (AshwiniChalla) [inline Tasks/ANT/ant.ps1]: Now this file is growing too much. Can we have a helpers class like maven ?

File: Tasks/ANT/ant.ps1 [hunk 2]
```diff
@@ -87,15 +87,20 @@ $reportDirectoryName = [guid]::NewGuid()
...
 $summaryFileName = "summary.xml"
+$summaryFileNameCobertura = "coverage.xml"
 $summaryFile = Join-Path $buildRootPath $reportDirectoryName 
 $summaryFile = Join-Path $summaryFile $summaryFileName
+$summaryFileCobertura = Join-Path $buildRootPath $reportDirectoryName
+$summaryFileCobertura = Join-Path $summaryFileCobertura $summaryFileNameCobertura
 # ensuring unique code coverage report task name by using guid
 $CCReportTask = "CodeCoverage_" +[guid]::NewGuid()
+# ensuring unique instrumentation task name by using guid
+$CCInstrumentTask = "CodeCoverageInstrument_" + [guid]::NewGuid()
```
Comment (AshwiniChalla) [inline Tasks/ANT/ant.ps1]: Cant we reuse the CCReportTask?

File: Tasks/ANT/task.json [hunk 3]
```diff
@@ -98,7 +98,8 @@
             "helpMarkDown": "Select the code coverage tool. Please make sure jacocoant.jar is available in lib folder of Ant installation.",
             "options": {
                 "None": "None",
-                "JaCoCo": "JaCoCo"
+                "JaCoCo": "JaCoCo",
+				"Cobertura": "Cobertura"
```
Comment (AshwiniChalla) [inline Tasks/ANT/task.json]: spaces

File: Tasks/Maven/task.json [hunk 4]
```diff
@@ -104,7 +104,8 @@
             "helpMarkDown": "Select the code coverage tool.",
             "options": {
                 "None": "None",
-                "JaCoCo": "JaCoCo"
+              "JaCoCo": "JaCoCo",
```
Comment (AshwiniChalla) [inline Tasks/Maven/task.json]: spaces all over

File: Tasks/Maven/task.json [hunk 5]
```diff
@@ -124,7 +125,7 @@
             "required": false,
             "groupName": "codeCoverage",
             "helpMarkDown": "This field is required for a multi module project. Code coverage reports are created using ANT task. Comma separated list of directories containing class files, archive files(jar, war etc.). Directories and archives are searched recursively for class files. Code coverage is reported for class files present in the directories. For example, classes1/Tests,classes2/Tests2. Refer http://www.eclemma.org/jacoco/trunk/doc/ant.html",
-            "visibleRule": "codeCoverageTool = JaCoCo"
```
Comment (AshwiniChalla) [inline Tasks/Maven/task.json]: This might be applicable only for jacoco. Recheck

File: Tasks/ANT/ant.ps1 (hunk 2 above)
Comment (prawalagarwal) [inline Tasks/ANT/ant.ps1]: These are used as target names. We need unique target names to invoke them later. 
We call instrument target first and then the report target.

File: Tasks/ANT/task.json [hunk 6]
```diff
@@ -98,7 +98,8 @@
             "helpMarkDown": "Select the code coverage tool. Please make sure jacocoant.jar is available in lib folder of Ant installation.",
```
Comment (AshwiniChalla) [inline Tasks/ANT/task.json]: Correct the help text. mention for jacoco jacocoant.jar is required

**Comment by AshwiniChalla** [comment]
Approved with comments

**Pull request merged**